drafts/df_exercises.md

```diff
@@ -0,0 +1,1044 @@
+By default, `head()` displays first five entries, but we have the option to passs it a specific number of rows we want to see.
+
+
+```python
+df.head(10)
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>SepalLength</th>
+      <th>SepalWidth</th>
+      <th>PetalLength</th>
+      <th>PetalWidth</th>
+      <th>Name</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>5.1</td>
+      <td>3.5</td>
+      <td>1.4</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>4.9</td>
+      <td>3.0</td>
+      <td>1.4</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>4.7</td>
+      <td>3.2</td>
+      <td>1.3</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>4.6</td>
+      <td>3.1</td>
+      <td>1.5</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>5.0</td>
+      <td>3.6</td>
+      <td>1.4</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>5.4</td>
+      <td>3.9</td>
+      <td>1.7</td>
+      <td>0.4</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>4.6</td>
+      <td>3.4</td>
+      <td>1.4</td>
+      <td>0.3</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>5.0</td>
+      <td>3.4</td>
+      <td>1.5</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>4.4</td>
+      <td>2.9</td>
+      <td>1.4</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>9</th>
+      <td>4.9</td>
+      <td>3.1</td>
+      <td>1.5</td>
+      <td>0.1</td>
+      <td>Iris-setosa</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Okay, now we know that our dataframe has five columns, that describe the properties of different species of irises.
+_But wait!_, you may ask, _there are 260–300 species of iris genus and there is no way they can all be represented in our 150 row dataframe!_. Well, my educated friend, you are absolutely right. It might be a good idea to take a look at all the _unique_ values we have in our __Name__ column.
+
+
+```python
+df.Name.unique()
+```
+
+
+
+
+    array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)
+
+
+
+We are going to look at the similarities and differences between the species later, but first, let's get familiar with each of them separately. For that, we can create a separate dataframe for each of the species.
+
+
+```python
+setosa = df[df.Name=='Iris-setosa']
```

Comment:
Oh, right. They just created Boolean arrays and then counted nonzero. My bad, I thought that came up somewhere before already. That's why I ask 'Is this a familiar behaviour?'. Showing them array indexing prior to this is a good idea.

I agree that your approach would be easier to understand if the notebook was passively read. But I wanted to try and create a more complicated line so that it actually requires them to stop for a second and think, maybe try to dissect it themselves. Teach them breaking up tasks to easier steps, hopefully. 
That is why I ask questions that suggest to try different parts of the line separately for themselves. Maybe that's too much to ask but I thought to give it a try. Can redo if you think it's too much.

As for `.loc`: that's where I think this would be too much new information for one line of code. But if we decide to break it into parts, then yeah, can add that, too.